### PR TITLE
Keep ENV['LOCAL_HTTPS'] with ApplicationControllerSpec (fix random fail)

### PR DESCRIPTION
**Pull request by ykzts**
This fixes the failure of the random Features Spec that we have suffered for a long time.

### error case

```console
$ ./bin/rspec ./spec/controllers/application_controller_spec.rb[1:1] ./spec/features/log_in_spec.rb[1:1] --seed 26878
[DEPRECATION] `have_enqueued_job` is deprecated.  Please use `have_enqueued_sidekiq_job` instead.
Run options: include {:ids=>{"./spec/controllers/application_controller_spec.rb"=>["1:1"], "./spec/features/log_in_spec.rb"=>["1:1"]}}                                                  

Randomized with seed 26878
                                                                                                                                                                                        
  1) Log in A valid email and password user is able to log in
     Failure/Error: expect(page).to have_css "div.app-holder"
       expected to find css "div.app-holder" but there were no matches
     # ./spec/features/log_in_spec.rb:17:in `block (2 levels) in <top (required)>'

 2/2 |============================================================================= 100 ==============================================================================>| Time: 00:00:01 

Finished in 1.54 seconds (files took 2.95 seconds to load)
2 examples, 1 failure

Failed examples:

rspec ./spec/features/log_in_spec.rb:11 # Log in A valid email and password user is able to log in

Randomized with seed 26878

Coverage report generated for RSpec to /Users/ykzts/works/ykzts/mastodon/coverage. 596 / 8516 LOC (7.0%) covered.
```

### how to find

<details>
<summary><code>$ ./bin/rspec --seed 26878 --bisect</code></summary>
<pre><code>Bisect started using options: "--seed 26878"
Running suite to find failures... (2 minutes 0.5 seconds)
Starting bisect with 1 failing example and 1084 non-failing examples.
Checking that failure(s) are order-dependent... failure appears to be order-dependent

Round 1: bisecting over non-failing examples 1-1084 .. ignoring examples 543-1084 (2 minutes 6.5 seconds)
Round 2: bisecting over non-failing examples 1-542 . ignoring examples 1-271 (27.38 seconds)
Round 3: bisecting over non-failing examples 272-542 .. ignoring examples 408-542 (34.6 seconds)
Round 4: bisecting over non-failing examples 272-407 . ignoring examples 272-339 (11.85 seconds)
Round 5: bisecting over non-failing examples 340-407 .. ignoring examples 374-407 (19.89 seconds)
Round 6: bisecting over non-failing examples 340-373 .. ignoring examples 357-373 (15.36 seconds)
Round 7: bisecting over non-failing examples 340-356 . ignoring examples 340-348 (6.97 seconds)
Round 8: bisecting over non-failing examples 349-356 .. ignoring examples 353-356 (16.2 seconds)
Round 9: bisecting over non-failing examples 349-352 .. ignoring examples 351-352 (14.42 seconds)
Round 10: bisecting over non-failing examples 349-350 . ignoring example 349 (7.94 seconds)
Bisect complete! Reduced necessary non-failing examples from 1084 to 1 in 4 minutes 48.1 seconds.

The minimal reproduction command is:
  rspec ./spec/controllers/application_controller_spec.rb[1:1] ./spec/features/log_in_spec.rb[1:1] --seed 26878</code></pre>
</details

### CODE REVIEW

**Review by akihikodaki — APPROVED** [review]

File: spec/features/log_in_spec.rb [hunk 1]
```diff
@@ -1,11 +1,14 @@
 require "rails_helper"
 
 feature "Log in" do
-  scenario "A valid email and password user is able to log in" do
-    email = "test@example.com"
-    password = "password"
+  given(:email)    { "test@examle.com" }
```
Comment (mjankowski) [inline spec/features/log_in_spec.rb]: Are these changes related to the ENV issue?

File: spec/controllers/application_controller_spec.rb [hunk 2]
```diff
@@ -37,20 +37,28 @@ def invalid_authenticity_token
...
-  it "does not force ssl if LOCAL_HTTPS is not 'true'" do
-    routes.draw { get 'success' => 'anonymous#success' }
-    ENV['LOCAL_HTTPS'] = ''
-    allow(Rails.env).to receive(:production?).and_return(true)
-    get 'success'
-    expect(response).to have_http_status(:success)
-  end
+  context do
+    around do |example|
+      original_local_https = ENV['LOCAL_HTTPS']
```
Comment (mjankowski) [inline spec/controllers/application_controller_spec.rb]: It looks like we have the climate control gem in the repo - https://github.com/thoughtbot/climate_control - but we are not using it in many places. Do you want to change this `around` to use climate control instead of the explicit before/after assignment here?

**Review by mjankowski — CHANGES_REQUESTED** [review]
Nice fix! I left a few comments, but this is close to ready.

File: spec/features/log_in_spec.rb (hunk 1 above)
Comment (ykzts) [inline spec/features/log_in_spec.rb]: I did a simple refactoring before identifying the cause. This is because I wanted to add cases and make it a particular help. It is a refactoring unrelated to this fix, should I undo it?

File: spec/controllers/application_controller_spec.rb (hunk 2 above)
Comment (ykzts) [inline spec/controllers/application_controller_spec.rb]: Oh! I didn't know that gem. looks good.

File: spec/features/log_in_spec.rb (hunk 1 above)
Comment (mjankowski) [inline spec/features/log_in_spec.rb]: No that's fine, just wanted to make sure it was here on purpose.

File: spec/controllers/application_controller_spec.rb (hunk 2 above)
Comment (ykzts) [inline spec/controllers/application_controller_spec.rb]: use this gem c585813.

**Review by mjankowski — APPROVED** [review]

**Comment by ykzts** [comment]
travis is down?
